kuka_rsi_hw_interface/src/rsi_command_handler.cpp

```diff
@@ -0,0 +1,325 @@
+      }
+      buffer_it++;
+      if (!element.CastParam(node_param, buffer_it)) {
+        auto i = std::sprintf(err_buf_, "Could not cast the ");
+        std::snprintf(err_buf_ + i, node_param.length_ + 1, "%s", node_param.data_ptr_);
+        i += node_param.length_;
+        std::sprintf(
+          err_buf_ + i, " param into the %s elements parameter list.",
+          element.GetName().c_str());
+        throw std::logic_error(err_buf_);
+      }
+      if (*buffer_it != '"') {
+        auto i = sprintf(err_buf_, "In \"");
+        std::snprintf(err_buf_ + i, node_param.length_ + 1, "%s", node_param.data_ptr_);
+        i += node_param.length_;
+        std::sprintf(err_buf_ + i, "\" param syntax error found");
+        throw std::logic_error(err_buf_);
+      }
+      buffer_it++;
+      numOfParam++;
+    }
+  }
+
+  // Fast forward to close bracket
+  for (; (size_t)(buffer_it - buffer) < buffer_size && *buffer_it != '>'; buffer_it++) {
+  }
+  if ((size_t)(buffer_it - buffer) >= buffer_size) {
+    throw std::range_error("Out of the buffers range");
+  }
+  buffer_it++;
+
+  // Could not find all parameter, or found too much
+  if (isBaseLessNode && numOfParam != element.Params().size()) {
+    std::sprintf(
+      err_buf_,
+      "%lu parameter found. It does not match with %s elements parameter list size.", numOfParam,
+      element.GetName().c_str());
+    throw std::logic_error(err_buf_);
+  }
+
+  // Node should have childs, but did not found any
+  if (isBaseLessNode && element.GetChilds().size() > 0) {
+    std::sprintf(
+      err_buf_, "%s node should have chields but did not found any",
+      element.GetName().c_str());
+    throw std::logic_error(err_buf_);
+  }
+  // fast forward if
+  for (; *buffer_it == ' '; buffer_it++) {
+  }
+  if ((size_t)(buffer_it - buffer) >= buffer_size) {
+    throw std::range_error("Out of the buffers range");
+  }
+
+  if (!isBaseLessNode) {
+    if (*buffer_it != '<') {
+      // Node base is data
+      if (!element.CastParam(node_name, buffer_it)) {
+        auto i = std::sprintf(err_buf_, "Could not cast the ");
+        std::snprintf(err_buf_ + i, node_name.length_ + 1, "%s", node_name.data_ptr_);
+        i += node_name.length_;
+        std::sprintf(
+          err_buf_ + i, " parameter into the %s elements parameter list",
+          element.GetName().c_str());
+        throw std::logic_error(err_buf_);
+      }
+      if (*buffer_it != '<') {
+        auto i = sprintf(err_buf_, "In \"");
+        std::snprintf(err_buf_ + i, node_name.length_ + 1, "%s", node_name.data_ptr_);
+        i += node_name.length_;
+        std::sprintf(err_buf_ + i, "\" param syntax error found");
+        throw std::logic_error(err_buf_);
+      }
+    } else {
+      // node base has childs
+      for (auto && child : element.Childs()) {
+        decodeNode(child, buffer, buffer_it, buffer_size);
+      }
+    }
+    for (; (size_t)(buffer_it - buffer) < buffer_size && *buffer_it != '<'; buffer_it++) {
```

Comment:
might be done later, but a lot of trash can come between the end tag and the next < tag here
e.g. hahaha&@{đ is valid according to the code
I think you were aware of this, but then this should be documented somewhere